src/analyses/base.ml

```diff
@@ -1105,19 +1110,32 @@ struct
       if Queries.Set.mem anyq asked then
         Queries.Result.top q (* query cycle *)
       else (
-        let asked' = Queries.Set.add anyq asked in
         match q with
-        | EvalInt e -> query_evalint (ask asked') gs st e (* mimic EvalInt query since eval_rv needs it *)
+        | EvalInt e -> query_evalint ~ctx e (* mimic EvalInt query since eval_rv needs it *)
```

Comment:
This no longer detects query cycles because the `asked` set will always be empty. I'll see if there's an example of doing this somewhere else.

Reply:
Here's an example of how to do it correctly: https://github.com/goblint/analyzer/blob/4f09c289cb2509b9e2e07bdb120ac7f34a2bcd59/src/analyses/base.ml#L1534-L1550.
Basically, `ctx` needs to be a function with the `asked` argument.